From 01e1434ad8cba2e7e7d66c8b2f87425738154970 Mon Sep 17 00:00:00 2001
From: Sam Bobrowicz <sam@elite-embedded.com>
Date: Thu, 11 Oct 2018 09:23:28 -0700
Subject: [PATCH 0467/1851] media: ov5640: Don't access ctrl regs when off

commit a067f43f4f770a260ba59187c0b0d112c20b6e0e from
https://github.com/Xilinx/linux-xlnx.git

Add a check to g_volatile_ctrl to prevent trying to read
registers when the sensor is not powered.

Signed-off-by: Sam Bobrowicz <sam@elite-embedded.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/i2c/ov5640.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/media/i2c/ov5640.c b/drivers/media/i2c/ov5640.c
index ee1ce5066bd4..cd837f8c188d 100644
--- a/drivers/media/i2c/ov5640.c
+++ b/drivers/media/i2c/ov5640.c
@@ -2573,6 +2573,13 @@ static int ov5640_g_volatile_ctrl(struct v4l2_ctrl *ctrl)
 
 	/* v4l2_ctrl_lock() locks our own mutex */
 
+	/*
+	 * If the sensor is not powered up by the host driver, do
+	 * not try to access it to update the volatile controls.
+	 */
+	if (sensor->power_count == 0)
+		return 0;
+
 	switch (ctrl->id) {
 	case V4L2_CID_AUTOGAIN:
 		val = ov5640_get_gain(sensor);
-- 
2.31.1

